src/components/metadata/list.js

```diff
@@ -16,6 +16,10 @@ class MetadataList extends React.PureComponent {
     return this.props.fields.length === 0
   }
 
+  get hasNewMetadataField() {
+    return this.props.edit != null && this.props.edit.id != null
```

Comment:
The logic here must be: `edit.id === field.id` (in practice a bit more complicated because id can be a number or an array so we can't just use `===`; we should maybe simplify that and always use an array, but we can do that as a second step) **and** `edit.property == null` (if there is a `property` we're editing an existing field).

Note that, we also need to update `isEditing()` and `edit()` belw accordingly.

Reply:
I added the new logic for `hasNewMetadataField`. I'm a bit lost on the last line of your comment.

All theses changes had fixed the initial bug, so now the new field component is shown on the 3 sections.

Reply:
We're changing the structure of the `state.edit.field` as part of this PR. If you look at the `edit()` method and the `isEditing()` predicate you'll notice that we had a slightly different structure (or just edit one of the values and observe the changes in `state.edit`. Basically, when you click to edit a value we want the structure to in `state.edit` to look something like `{ id: [24], property: 'http://....' }` -- in other words, editing and adding a field should use the same structure (when you're about to add a new field, the property is just empty).

This also sets us up for editing other things about a field than just the value (datatype or language) in the future.